plugins/macos_logs.yaml

```diff
@@ -0,0 +1,91 @@
+version: 0.0.1
+title: MacOS
+description: Log parser for MacOS
+parameters:
+  - name: enable_system_log
+    type: bool
+    default: true
+  - name: system_log_path
+    type: "[]string"
+    default:
+      - "/var/log/system.log"
+  - name: enable_install_log
+    type: bool
+    default: true
+  - name: install_log_path
+    type: "[]string"
+    default: 
+      - "/var/log/install.log"
+  - name: start_at
+    type: string
+    supported:
+      - beginning
+      - end
+    default: end
+template: |
+  receivers:
+    {{ if .enable_system_log }}
+    filelog/system_logs:
+      include: 
+        {{ range $fp := .system_log_path }}
+        - '{{ $fp }}'
+        {{ end }}
+      multiline: 
+        line_start_pattern: '\w{3}\s*\d{1,2} \d{2}:\d{2}:\d{2}'
+      start_at: {{ .start_at }}
+      attributes:
+        log_type: 'macos.system'
+      operators:
+        - type: regex_parser
+          regex: '^(?P<timestamp>\w{3}\s*\d{1,2} \d{2}:\d{2}:\d{2}) (---|(?P<host>[^ ]*))? ((?P<process>[^\[]*)\[(?P<pid>[^\]]*)\])?( \((?P<subprocess>[^\[]*)(\[(?P<spid>[^\]]*)\])?\))?(: )?(?P<message>[\w\W]*)'
+          timestamp:
+            parse_from: attributes.timestamp
+            layout_type: gotime
+            layout: 'Jan _2 15:04:05'
+    {{ end }} ## .enable_system_log
+
+    {{ if .enable_install_log }}
+    filelog/install_logs:
+      include: 
+        {{ range $fp := .install_log_path }}
+        - '{{ $fp }}'
+        {{ end }}
+      multiline: 
+        line_start_pattern: '^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}[+-]\d{2}|^\w{3}\s*\d{1,2} \d{2}:\d{2}:\d{2}'
+      start_at: {{ .start_at }}
+      attributes:
+        log_type: 'macos.install'
+      operators:
+        - id: id_router
+          type: router
+          routes:
+            - output: time_parser_with_timezone
+              expr: $ matches '^\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}[+-]\\d{2}'
+            - output: time_parser_without_timezone
+              expr: $ matches '^\\w{3}\\s*\\d{1,2} \\d{2}:\\d{2}:\\d{2}'
+        - id: time_parser_with_timezone
+          type: regex_parser
+          regex: '^(?P<timestamp>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}[+-]\d{2}) (---|(?P<host>[^ ]*))? ((?P<process>[^\[]*)\[(?P<pid>[^\]]*)\])?( \((?P<subprocess>[^\[]*)(\[(?P<spid>[^\]]*)\])?\))?(: )?(?P<message>[\w\W]*)'
+          timestamp:
+            parse_from: attributes.timestamp
+            layout_type: gotime
+            layout: '2006-01-02 15:04:05-07'
+        - id: time_parser_without_timezone
+          type: regex_parser
+          regex: '^(?P<timestamp>\w{3}\s*\d{1,2} \d{2}:\d{2}:\d{2}) (---|(?P<host>[^ ]*))? ((?P<process>[^\[]*)\[(?P<pid>[^\]]*)\])?( \((?P<subprocess>[^\[]*)(\[(?P<spid>[^\]]*)\])?\))?(: )?(?P<message>[\w\W]*)'
+          timestamp:
+            parse_from: attributes.timestamp
+            layout_type: gotime
+            layout: 'Jan _2 15:04:05'
```

Comment:
I think you need and add processor here just to point the two timestamp blocks back to a single point. See [mysql](https://github.com/observIQ/observiq-otel-collector/blob/main/plugins/mysql_logs.yaml#L634-L637) as example.

Reply:
fixed